mesop/component_helpers/helper.py

```diff
@@ -227,6 +227,21 @@ def insert_composite_component(
   )
 
 
+def insert_custom_component(
+  component_name: str,
+  proto: Message,
+  key: str | None = None,
+  style: Style | None = None,
+):
+  return insert_component(
+    # Prefix with <custom> to ensure there's never any overlap.
+    type_name="<custom>" + component_name,
```

Comment:
Question, so when someone is using the custom component. Say the mic component. Would they still do something like me.mic()?

Reply:
No, they would import the mic component from their module, e.g.

```py
from my_package.mic_component import mic
```

Btw, the type_name here is used internally to map between Python and JS, but it doesn't actually need to correspond with the component function name, etc. (we keep it the same for consistency).